pallets/payment-streams/src/lib.rs

```diff
@@ -447,8 +448,15 @@ pub mod pallet {
             // Check that the extrinsic was executed by the root origin
             ensure_root(origin)?;
 
+            let current_tick = Self::get_current_tick();
+
             // Execute checks and logic, update storage
-            Self::do_update_fixed_rate_payment_stream(&provider_id, &user_account, new_rate)?;
+            Self::do_update_fixed_rate_payment_stream(
+                &provider_id,
+                &user_account,
+                new_rate,
+                current_tick,
+            )?;
```

Comment:
Regardless, it shouldn't be the `current_tick` that is used here, but rather the tick up to which this payment stream was charged, which is not necessarily the same. In fact in most cases, it will differ. For BSPs for instance, it is the last tick for which they submitted a proof for.